crates/edr_eth/src/remote/client.rs

```diff
@@ -636,24 +637,26 @@ impl RpcClient {
     /// Whether the block number should be cached based on its depth.
     pub async fn is_cacheable_block_number(
         &self,
-        block_number: &U256,
+        block_number: u64,
     ) -> Result<bool, RpcClientError> {
         let chain_id = self.chain_id().await?;
         let latest_block_number = self.cached_block_number().await?;
 
         Ok(is_safe_block_number(IsSafeBlockNumberArgs {
-            chain_id: &chain_id,
-            latest_block_number: &latest_block_number,
+            chain_id,
+            latest_block_number,
             block_number,
         }))
     }
 
     /// Calls `eth_chainId` and returns the chain ID.
-    pub async fn chain_id(&self) -> Result<U256, RpcClientError> {
+    pub async fn chain_id(&self) -> Result<u64, RpcClientError> {
         let chain_id = *self
             .chain_id
             .get_or_try_init(|| async {
-                self.call_without_cache(MethodInvocation::ChainId()).await
+                self.call_without_cache::<U64>(MethodInvocation::ChainId())
+                    .await
+                    .map(|chain_id| chain_id.as_limbs()[0])
```

Comment:
I think `try_into` should work here.

Reply:
It's infallible though. So we'd be unnecessarily wrapping it in a result and then calling expect